src/ape/managers/project/manager.py

```diff
@@ -368,7 +368,18 @@ def __getattr__(self, attr_name: str) -> Union[ContractContainer, ContractNamesp
 
             # Fixes anomaly when accessing non-ContractType attributes.
             # Returns normal attribute if exists. Raises 'AttributeError' otherwise.
-            return self.__getattribute__(attr_name)  # type: ignore
+            try:
+                return self.__getattribute__(attr_name)  # type: ignore
+            except AttributeError as err:
+                message = f"ProjectManager has no attribute or contract named '{attr_name}'."
+                missing_exts = self.extensions_with_missing_compilers([])
+                if missing_exts:
+                    message = (
+                        f"{message} Could it be from one of the missing compilers for extensions:"
+                        + f"{', '.join(missing_exts)}"
```

Comment:
Slight tweak to something I just noticed - you could use a space here after the colon and you can remove the f-string from the second value:

```python
message = (
    f"{message} Could it be from one of the missing compilers for extensions: "
    + ", ".join(missing_exts)
)
```

Reply:
In general, I prefer f-str over string addition. It is more readable imo